### PR TITLE
Slideseq

### LINKED ISSUE #428: Create metadata schema for a new assay: SLIDEseq
Here's a link to the spreadsheet for SLIDEseq metadata:
https://docs.google.com/spreadsheets/d/1KDBigt6Kqi7VRFb_uc3zkc17tedi4gz_oS0Y2T0fi08/edit?usp=sharing
 It's much the same as RNAseq except for a new puck_id column.

### PR DESCRIPTION
**Pull request by mccalluc**
Fix #428.

@cebriggs7135 :
- Description for puck_id; please provide as a suggestion in github.
- Confirm that the `sc` fields can be dropped.

### CODE REVIEW

File: docs/slideseq/README.md [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+
+### `library_adapter_sequence`
+Adapter sequence to be used for adapter trimming.
+
+| constraint | value |
+| --- | --- |
+| required | `True` |
+
+### `puck_id`
+TODO.
```
Comment (cebriggs7135) [inline docs/slideseq/README.md:219]:
``` suggestion
Slide-seq captures RNA sequence data on spatially barcoded arrays of beads. Beads are fixed to a slide in a region shaped like a round puck. Each puck has a unique puck_id.
```

**Review by cebriggs7135 — COMMENTED** [review]
Let me check with the SLIDEseq team to see if they agree that "cell" should be changed to "bead" throughout the schema.

**Comment by mccalluc** [comment]
> Let me check with the SLIDEseq team to see if they agree that the 3 instances of "cell_barcode" should be changed to "bead_barcode"in the schema.

Sure, if that is more accurate... but if it's just about the names, I have a preference for consistency across the assay types... but obviously consistency shouldn't trump correctness. Thanks.

File: docs/slideseq/README.md [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+[`acquisition_instrument_vendor`](#acquisition_instrument_vendor)<br>
+[`acquisition_instrument_model`](#acquisition_instrument_model)<br>
+[`rnaseq_assay_input`](#rnaseq_assay_input)<br>
+[`rnaseq_assay_method`](#rnaseq_assay_method)<br>
+[`library_construction_protocols_io_doi`](#library_construction_protocols_io_doi)<br>
+[`library_layout`](#library_layout)<br>
+[`library_adapter_sequence`](#library_adapter_sequence)<br>
+[`puck_id`](#puck_id)<br>
+[`is_technical_replicate`](#is_technical_replicate)<br>
+[`cell_barcode_read`](#cell_barcode_read)<br>
```
Comment (cebriggs7135) [inline docs/slideseq/README.md:40]:
```suggestion
[`bead_barcode_read`](#cell_barcode_read)<br>
```

File: docs/slideseq/README.md [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+[`cell_barcode_offset`](#cell_barcode_offset)<br>
```
Comment (cebriggs7135) [inline docs/slideseq/README.md:41]:
```suggestion
[`bead_barcode_offset`](#cell_barcode_offset)<br>
```

File: docs/slideseq/README.md [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+[`cell_barcode_size`](#cell_barcode_size)<br>
```
Comment (cebriggs7135) [inline docs/slideseq/README.md:42]:
```suggestion
[`bead_barcode_size`](#cell_barcode_size)<br>
```

File: docs/slideseq/README.md [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+
+### `is_technical_replicate`
+Is the sequencing reaction run in repliucate, TRUE or FALSE.
+
+| constraint | value |
+| --- | --- |
+| type | `boolean` |
+| required | `True` |
+
+### `cell_barcode_read`
```
Comment (cebriggs7135) [inline docs/slideseq/README.md:233]:
```suggestion
### `bead_barcode_read`
```

File: docs/slideseq/README.md [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+Which read file contains the cell barcode.
+
+| constraint | value |
+| --- | --- |
+| required | `True` |
+
+### `cell_barcode_offset`
```
Comment (cebriggs7135) [inline docs/slideseq/README.md:240]:
```suggestion
### `bead_barcode_offset`
```

File: docs/slideseq/README.md [hunk 7]
```diff
@@ -0,0 +1,340 @@
...
+Position(s) in the read at which the cell barcode starts.
+
+| constraint | value |
+| --- | --- |
+| required | `True` |
+
+### `cell_barcode_size`
```
Comment (cebriggs7135) [inline docs/slideseq/README.md:247]:
```suggestion
### `bead_barcode_size`
```

File: src/ingest_validation_tools/table-schemas/level-2/slideseq.yaml [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+    description: Adapter sequence to be used for adapter trimming
+  -
+    name: puck_id
+    description: TODO
+  -
+    name: is_technical_replicate
+    description: Is the sequencing reaction run in repliucate, TRUE or FALSE
+    type: boolean
+  -
+    name: cell_barcode_read
```
Comment (cebriggs7135) [inline src/ingest_validation_tools/table-schemas/level-2/slideseq.yaml:50]:
```suggestion
    name: bead_barcode_read
```

File: src/ingest_validation_tools/table-schemas/level-2/slideseq.yaml [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+    description: Which read file contains the cell barcode
```
Comment (cebriggs7135) [inline src/ingest_validation_tools/table-schemas/level-2/slideseq.yaml:51]:
```suggestion
    description: Which read file contains the bead barcode
```

File: src/ingest_validation_tools/table-schemas/level-2/slideseq.yaml [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+  -
+    name: cell_barcode_offset
```
Comment (cebriggs7135) [inline src/ingest_validation_tools/table-schemas/level-2/slideseq.yaml:53]:
```suggestion
    name: bead_barcode_offset
```

File: src/ingest_validation_tools/table-schemas/level-2/slideseq.yaml [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+    description: Position(s) in the read at which the cell barcode starts.
```
Comment (cebriggs7135) [inline src/ingest_validation_tools/table-schemas/level-2/slideseq.yaml:54]:
```suggestion
    description: Position(s) in the read at which the bead barcode starts.
```

File: src/ingest_validation_tools/table-schemas/level-2/slideseq.yaml [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+  -
+    name: cell_barcode_size
```
Comment (cebriggs7135) [inline src/ingest_validation_tools/table-schemas/level-2/slideseq.yaml:56]:
```suggestion
    name: bead_barcode_size
```

File: src/ingest_validation_tools/table-schemas/level-2/slideseq.yaml [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+    description: Length of the cell barcode in base pairs
```
Comment (cebriggs7135) [inline src/ingest_validation_tools/table-schemas/level-2/slideseq.yaml:57]:
```suggestion
    description: Length of the bead barcode in base pairs
```

File: src/ingest_validation_tools/table-schemas/level-2/slideseq.yaml [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+  -
+    heading: Level 2
+    name: acquisition_instrument_vendor
+    description: An acquisition_instrument is the device that contains the signal detection hardware and signal processing software. Assays can generate signals such as light of various intensities or color or signals representing molecular mass.
+  -
+    name: acquisition_instrument_model
+    description: Manufacturers of an acquisition instrument may offer various versions (models) of that instrument with different features or sensitivities. Differences in features or sensitivities may be relevant to processing or interpretation of the data.
+  -
+    name: rnaseq_assay_input
+    description: Number of cell/nuclei input to the assay
```
Comment (cebriggs7135) [inline src/ingest_validation_tools/table-schemas/level-2/slideseq.yaml:29]:
```suggestion
 delete
```

**Review by cebriggs7135 — CHANGES_REQUESTED** [review]

**Comment by mccalluc** [comment]
@cebriggs7135 : Reminder that the `.md` files are generated from the `.yaml`... So usually, just make suggestions on the `.yaml`, and then I'll refresh the `.md`: Should be less work on your end.

File: src/ingest_validation_tools/table-schemas/level-2/slideseq.yaml (hunk 14 above)
Comment (mccalluc) [reply src/ingest_validation_tools/table-schemas/level-2/slideseq.yaml:29]: @cebriggs7135 : For deletions, select the entire range you want deleted, and delete all the lines in the suggestion block, so the closing <code>&#x60;&#x60;&#x60;</code> immediately follows the opening <code>&#x60;&#x60;&#x60;</code>.

File: src/ingest_validation_tools/table-schemas/level-2/slideseq.yaml [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+  -
```
Comment (mccalluc) [inline src/ingest_validation_tools/table-schemas/level-2/slideseq.yaml:30]:
```suggestion
  -
```

File: src/ingest_validation_tools/table-schemas/level-2/slideseq.yaml (hunk 14 above)
Comment (mccalluc) [inline src/ingest_validation_tools/table-schemas/level-2/slideseq.yaml:29]:
```suggestion
```

Comment (mccalluc) [reply src/ingest_validation_tools/table-schemas/level-2/slideseq.yaml]: resolved separately.

**Comment by mccalluc** [comment]
@cebriggs7135 : almost there! Last bit: what assay type should be used? 
```
    name: assay_type
    constraints:
      enum: [] # TODO
```

File: src/ingest_validation_tools/table-schemas/level-2/slideseq.yaml [hunk 16]
```diff
@@ -0,0 +1,87 @@
...
+    name: is_technical_replicate
+    description: Is the sequencing reaction run in repliucate, TRUE or FALSE
+    type: boolean
+  -
+    name: bead_barcode_read
+    description: Which read file contains the bead barcode
+  -
+    name: bead_barcode_offset
+    description: Position(s) in the read at which the bead barcode starts.
```
Comment (cebriggs7135) [inline src/ingest_validation_tools/table-schemas/level-2/slideseq.yaml:52]:
```suggestion
    description: Position(s) in the read at which the bead barcode starts
```

**Review by cebriggs7135 — CHANGES_REQUESTED** [review]
I just removed one period at the end of a description. Otherwise, this is done.

**Pull request merged**